Update docs for new BUF_TOKEN format (#343)
bufbuild/docs.buf.build: docs/bsr/authentication.md

### BUF_TOKEN

`BUF_TOKEN` is an environment variable that holds the API token, used for authentication.

> #### Default Token
> 
> `BUF_TOKEN` can contain only a default token. This token is used when authenticating with
> every remote address. You can specify at most one default token in the environment variable.
> You can set `BUF_TOKEN` by:
> ```terminal
> $ export BUF_TOKEN=${DEFAULT_TOKEN}
> ```

Reviewer: Let's drop "default" with "single" since default is also the empty token. And I think we can drop the "at most one" since "single" implies that. "remote address" is also a little in-the-weeds, so maybe "BSR" is sufficient here.

Reads like this:

```suggestion
#### Single Token

`BUF_TOKEN` can contain a single token. This token is used when authenticating with
the BSR.
You can set `BUF_TOKEN` by:
```terminal
$ export BUF_TOKEN=${DEFAULT_TOKEN}
```
```

> #### Multiple BSR Remote Tokens
> 
> The `BUF_TOKEN` environment can also contain tokens with specified remote address. You can
> set `BUF_TOKEN` by:
> ```terminal
> $ export BUF_TOKEN=${USER1}:${TOKEN1}@{REMOTE1},${USER2}:${TOKEN2}@{REMOTE2},...,{DEFAULT_TOKEN}
> ```
> The default token here is used when authenticating with remote addresses when a matching remote
> token is not provided. The default token is not required.
> 
> You can have multiple remote Buf Tokens in the format `${USER}:${BUF_TOKEN}@${REMOTE_ADDRESS}`
> separated by `,`. The `buf` CLI will find the correct authentication token for different
> remote address.

Reviewer: And now with multiple tokens we can introduce the concept of a default token. Small wording nits and ordering changes below.

```suggestion
#### Multiple Tokens

The `BUF_TOKEN` environment can also contain tokens with specified BSR
addresses. Each token has the format `${USER}:${BUF_TOKEN}@${REMOTE_ADDRESS}`
with multiple tokens separated by `,`. The `buf` CLI will find the correct
authentication token for different remote address. You can set `BUF_TOKEN` by:
<backticks>terminal
$ export BUF_TOKEN=${USER1}:${TOKEN1}@{REMOTE1},${USER2}:${TOKEN2}@{REMOTE2},...,{DEFAULT_TOKEN}
<backticks>

The default token is used when authenticating with remote addresses when a
matching remote token is not provided. The default token is not required.
```

> Note that we have [official GitHub Actions](../ci-cd/github-actions.md) that
> enable you to quickly configure authentication for CI jobs.

> ## Multiple BSR Remotes

Reviewer: Fold this information into the BUF_TOKEN section (a single location for all token information). Start off with an example for the default single-token case. Then expand with multiple tokens using what you have below.

> If you are working with multiple BSR remotes, you can set the all the Buf tokens in the
> `BUF_TOKEN` environment variable. The `BUF_TOKEN` environment variable also takes the format:

Reviewer: It's easier to read a statement than starting with a precondition.

```suggestion
You can set the all the Buf tokens in the `BUF_TOKEN` environment variable when working with multiple BSR remotes. The `BUF_TOKEN` environment variable takes the format:
```

```
${USER1}:${BUF_TOKEN1}@{REMOTE_ADDRESS1},${USER2}:${BUF_TOKEN2}@{REMOTE_ADDRESS2},...,{DEFAULT_TOKEN}
```
You can have multiple remote Buf Tokens in the format `${USER}:${BUF_TOKEN}@${REMOTE_ADDRESS}`

> separated by `,`. The correct token will be used for authentication for the corresponded
> remote address. The `${DEFAULT_TOKEN}` is optional. You can specify at most one default

Reviewer: Simplify wording and break this multiple and default sections into different paragraphs.

```suggestion
separated by `,`. The token used for authentication is the token with the matching remote address.

The `${DEFAULT_TOKEN}` is optional. You can specify at most one default
```

> tokens in the environment variable. The default token is always used when authenticating
> with remote addresses that a specific token is not provided.

Reviewer: This info should be at the very top; it describes what a default token is for.

```suggestion
tokens in the environment variable. The default token is used when authenticating
with remote addresses when a matching remote token is not provided.
```
